deepspeed/__init__.py

```diff
@@ -12,10 +12,14 @@
 from torch.optim.lr_scheduler import _LRScheduler
 from packaging import version as pkg_version
 
-try:
-    import triton  # noqa: F401 # type: ignore
-    HAS_TRITON = True
-except ImportError:
+# Skip Triton import for AMD due to pytorch-triton-rocm module breaking device API in DeepSpeed
+if 'AMD' not in torch.cuda.get_device_name():
+    try:
+        import triton  # noqa: F401 # type: ignore
+        HAS_TRITON = True
+    except ImportError:
+        HAS_TRITON = False
+else:
     HAS_TRITON = False
```

Comment:
```suggestion
from deepspeed.accelerator import get_accelerator

# Skip Triton import for AMD due to pytorch-triton-rocm module breaking device API in DeepSpeed
HAS_TRITON = False

if not get_accelerator().is_rocm_device():
    try:
        import triton  #  noqa: F401 # type: ignore
        HAS_TRITON = True
    except ImportError:
        pass
```

Because of the accelerator abstraction, we cannot use `torch.cuda` calls directly. I recommend expanding the accelerator to have a `is_rocm_device` method and using it here.